db/sql/05_msar.sql

```diff
@@ -2906,13 +2906,15 @@ Args:
 */
 DECLARE
   con_create_defs __msar.con_def[];
+  added_contype char[];
 BEGIN
   con_create_defs := __msar.process_con_def_jsonb(tab_id, con_defs);
   PERFORM __msar.add_constraints(
     __msar.get_qualified_relation_name(tab_id),
     variadic con_create_defs
   );
-  RETURN array_agg(oid) FROM pg_constraint WHERE conrelid=tab_id;
+  added_contype := array_agg(x->>'type') FROM jsonb_array_elements(con_defs) AS x;
+  RETURN array_agg(oid) FROM pg_constraint WHERE conrelid=tab_id AND contype=ANY(added_contype);
```

Comment:
This changes the semantics of the return value, which changes the response at the API. We used to return all constraints for the table via the API, but this now only returns constraints which happen to have one of the same constraint types as one of the added constraints. I don't think that change makes sense.

Reply:
Reverted, I thought we were returning only the constraints we added.